jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileToolkitCredentialsProvider.kt

```diff
@@ -103,17 +104,19 @@ class ProfileToolkitCredentialsProvider(
                     AwsClientManager.userAgent
                 )
 
-                StsAssumeRoleCredentialsProvider.builder()
-                    .stsClient(stsClient)
-                    .refreshRequest(Supplier {
-                        createAssumeRoleRequest(
-                            mfaSerial,
-                            roleArn,
-                            roleSessionName,
-                            externalId
-                        )
-                    })
-                    .build()
+                CorrectThreadCredentialsProvider(
```

Comment:
Is this the only place we want to use this wrapper?

Reply:
Today this is only call. I originally tried to do it with wrapping all ToolkitCredentialProviders but that isnt easy since that lives in core, and not jetbrains-core...

Reply:
k